File: Bilets/Singleton.cs
// Singleton pattern realisation 
class YabaiBeatricheF33t
{
    //Singleton assumes that class will have only one instance
    //So let`s create class that defines me, that is YabaiBeatricheF33t
    private static YabaiBeatricheF33t instance;
    //This is varable that contains an instance of the class, so it`s contains me 
    private YabaiBeatricheF33t() {}
    //Now we create class constructor 
    public static YabaiBeatricheF33t getInstaince()
    {
        if (instance == null)
        {
            instance = new YabaiBeatricheF33t();
        }
        return instance;
    }
    //So we create a method that will create and return an instance of the class, if this instance does not exist or just return already existing instance 
}

File: lab_3/singleton.py
class Singleton:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super().__new__(cls)
            cls._instance.value = 0
        return cls._instance

if __name__ == '__main__':
    a = Singleton()
    b = Singleton()

    # проверяем, что оба объекта — один и тот же экземпляр
    print(a is b)

    # проверяем глобальный доступ
    a.value = 1
    print(b.value)